site/src/pages/ManagementSettingsPage/IdpSyncPage/IdpSyncPageView.tsx

```diff
@@ -19,22 +23,57 @@ import {
 	TableLoaderSkeleton,
 	TableRowSkeleton,
 } from "components/TableLoader/TableLoader";
+import { TabLink, Tabs, TabsList } from "components/Tabs/Tabs";
 import type { FC } from "react";
+import { useSearchParams } from "react-router-dom";
 import { MONOSPACE_FONT_FAMILY } from "theme/constants";
 import { docs } from "utils/docs";
+import { ExportPolicyButton } from "./ExportPolicyButton";
+import { IdpPillList } from "./IdpPillList";
 
-export type IdpSyncPageViewProps = {
-	oidcConfig: OIDCConfig | undefined;
-};
+interface IdpSyncPageViewProps {
+	groupSyncSettings: GroupSyncSettings | undefined;
+	roleSyncSettings: RoleSyncSettings | undefined;
+	groups: Group[] | undefined;
+	organization: Organization;
+}
+
+export const IdpSyncPageView: FC<IdpSyncPageViewProps> = ({
+	groupSyncSettings,
+	roleSyncSettings,
+	groups,
+	organization,
+}) => {
+	const [searchParams] = useSearchParams();
+	const groupsMap = new Map<string, string>();
+	if (groups) {
+		for (const group of groups) {
+			groupsMap.set(group.id, group.display_name || group.name);
+		}
+	}
```

Comment:
Would it make sense to lift this up to the page component? I don't expect this map to change all that often, and putting it there should limit re-renders more without having to resort to `useMemo`